### PR TITLE
Monitor & Metrics Draft

### PR DESCRIPTION
**Pull request by stwerner97**
So far, the draft only re-factors how metrics are tracked and provides some defaults. Now, scalar, per user equipment and per basestation metrics can easily be defined and added to the monitor at runtime.

### CODE REVIEW

**Comment by stefanbschneider** [comment]
Do you think you could complete and merge this PR by the end of the month? And ideally, add some example on how to use the metrics? Eg, include them in the tutorial notebook: https://github.com/stefanbschneider/mobile-env/blob/main/examples/demo.ipynb

If it's too much work, we can also drop it or keep the PR open - no worries.

**Comment by stwerner97** [comment]
Sorry for the late response. Sure, I'll draft the changes in the next days.

**Comment by stwerner97** [comment]
Hallo Stefan,

nochmal sorry für die späte Antwort! Ich war nach den Ferien erstmal
ziemlich mit der Masterarbeit beschäftigt 😅

Viele Grüße
Stefan

**Review by stefanbschneider — APPROVED** [review]
Great, thank you! 

Did you change anything in the demo notebook? I couldn't really find any difference in the notebook and the diff does not work well for notebooks. Either way, there seems to be a merge conflict.

It would be great if you could provide some small, simple example code that illustrates how to monitor and retrieve metrics for evaluation with mobile-env.

If you update the notebook, could you also add a [link to the GitHub repo](https://github.com/stefanbschneider/mobile-env) at the beginning in the first markdown block of the notebook? Since the demo paper directly links to the notebook in Google CoLab, there should be a link inside the notebook back to the mobile-env repo.

Thanks again!

**Comment by stefanbschneider** [comment]
Btw, could you please bump the version to v1.1.0 and make a new release once this is merged? The release pushes the changes to PyPI, which is important since the demo notebook installs mobile-env from PyPI.

**Comment by stwerner97** [comment]
An example on how to use monitoring with `mobile-env` should already be included in the notebook. 

Sure, I will add a link to the notebook and update the PyPi version.

File: mobile_env/core/base.py [hunk 1]
```diff
@@ -83,8 +87,9 @@ def __init__(self, stations, users, config={}):
         self.conn_isolines = None
         self.mb_isolines = None
 
-        # tracks performance metrics for the on-going simulation
-        self.metrics = None
+        # add metrics required for visualization to configured metrics & set up monitor
+        config["metrics"]["scalar_metrics"].update({"number connections": metrics.number_connections, "number connected": metrics.number_connected, "mean utility": metrics.mean_utility, "mean datarate": metrics.mean_datarate})
```
[the remark this answers is not in the document]
Comment (stwerner97) [reply mobile_env/core/base.py]: I think so. The latest commit should resolve the issues, though :)

**Pull request merged**